Fcom 211

hodorsoft opened the pull request:
.netstandard + csv & file output support (hox, breaking changes due to task settings arragement)

RedBraces committed:
FCOM-204: fixes for how Oracle decimals are handled when converting results to JSON

RedBraces committed:
FCOM-204: proper handling for NULL values in decimal column types

RedBraces committed:
FCOM-204: marked test as Ignore because it requires a local Oracle DB

Commit:
FCOM-211: .netstandard2.0 version and breaking changes
- more user friends task configuration
- added csv output as Options
- all output types can now be written directly into a file

Commit:
FCOM-211: some parameter arrangements for task + fixed a typo in one dependency

Commit:
FCOM-211: updated README.md

narnil commented on Frends.Community.Oracle.Query/Extensions.cs:
```diff
@@ -16,35 +20,230 @@ public static TEnum ConvertEnum<TEnum>(this Enum source)
+        /// <param name="command"></param>
+        /// <param name="output"></param>
+        /// <returns></returns>
+        public static string ToXml(this OracleCommand command, OutputProperties output)
```
This is not used anywhere and should be removed.

narnil commented on Frends.Community.Oracle.Query/Frends.Community.Oracle.Query.csproj:
```diff
@@ -1,63 +1,31 @@
-    <DocumentationFile>bin\Release\Frends.Community.Oracle.Query.xml</DocumentationFile>
+
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
+    <DocumentationFile>C:\Users\raustvi\Documents\duunit\Frends.Community.Tasks\Frends.Community.Oracle.Query\Frends.Community.Oracle.Query\Frends.Community.Oracle.Query.xml</DocumentationFile>
```
Hard coded value pointing to user folder should be changed to use project folder dynamically.

narnil commented on Frends.Community.Oracle.Query/Extensions.cs:
```diff
@@ -16,35 +20,230 @@ public static TEnum ConvertEnum<TEnum>(this Enum source)
+                    writer.Culture = culture;
+
+                    // start array
+                    await writer.WriteStartArrayAsync();
```
cancellationToken should be passed to async operations.

narnil commented on Frends.Community.Oracle.Query/QueryTask.cs:
```diff
@@ -14,54 +14,50 @@ public class QueryTask
                     try
                     {
-                        connection.Open();
+                        await c.OpenAsync();
```
Pass cancellationToken.

Commit:
FCOM-211: changes due to review comments.
- CancellationToken is passed more often, especially in ToJsonAsync(..) where Newtonsoft supports its usage with method calls
- removed unused ToXml(..) extension method
- removed user specific filepaths from Frends.Community.Oracle.Query.csproj related to xml documentation
- nuspec package version updated to 2.0.1

hodorsoft commented:
Requested changes have been made to the same branch
